src/project.rs

```diff
@@ -213,3 +226,35 @@ pub fn collect_source(
     }
     Ok(())
 }
+
+fn check_app_file_version_matches(
+    root: impl AsRef<Path>,
+    project_config: &PackageConfig,
+) -> Result<(), Error> {
+    let mut app_src_path = root.as_ref().to_path_buf();
+    app_src_path.push("src");
+    app_src_path.push(format!("{}.app.src", &project_config.name));
+
+    let contents = std::fs::read_to_string(&app_src_path)
+        .map_err(|err| Error::FileIO {
+            action: FileIOAction::Read,
+            kind: FileKind::File,
+            err: Some(err.to_string()),
+            path: app_src_path.clone(),
+        })?
+        // Remove all new lines so we can regex easily across the content
+        .replace("\n", "");
+
+    // Search for the vsn line & extract the text in double quotes
+    let re = Regex::new("\\{ *vsn *, *\"([^\"]*)\" *\\}").unwrap();
```

Comment:
Can we use `.gleam_expect` here instead of `unwrap`? It produces a better error message to the user.

Reply:
Done.